app/navigation/router.js

```diff
@@ -180,39 +189,49 @@ class Router extends React.Component {
         const {modalVisible} = this.props;
 
         return (
-            <Drawer
-                open={leftDrawerOpen}
-                type='displace'
-                disabled={modalVisible}
-                content={leftDrawerContent}
-                tapToClose={true}
-                openDrawerOffset={0.2}
-                onRequestClose={this.props.actions.closeDrawers}
-                panOpenMask={0.1}
-                panCloseMask={0.2}
-                panThreshold={0.2}
-                acceptPan={navigationProps.allowSwipe}
-                negotiatePan={true}
-            >
+            <View style={{flex: 1}}>
                 <Drawer
-                    open={rightDrawerOpen}
+                    open={leftDrawerOpen}
                     type='displace'
-                    side='right'
                     disabled={modalVisible}
-                    content={rightDrawerContent}
+                    content={leftDrawerContent}
                     tapToClose={true}
                     openDrawerOffset={0.2}
                     onRequestClose={this.props.actions.closeDrawers}
+                    panOpenMask={0.1}
+                    panCloseMask={0.2}
+                    panThreshold={0.2}
+                    acceptPan={navigationProps.allowSwipe}
+                    negotiatePan={true}
+                    useInteractionManager={true}
                 >
+                    <Drawer
+                        open={rightDrawerOpen}
+                        type='displace'
+                        side='right'
+                        disabled={modalVisible}
+                        content={rightDrawerContent}
+                        tapToClose={true}
+                        openDrawerOffset={0.2}
+                        onRequestClose={this.props.actions.closeDrawers}
+                    >
+                        <NavigationExperimental.Transitioner
+                            style={{flex: 1}}
+                            navigationState={this.props.navigation}
+                            render={this.renderTransition}
+                            configureTransition={this.configureTransition}
+                        />
+                    </Drawer>
+                </Drawer>
+                <NavigationModal show={isModal}>
```

Comment:
1. If we don't always render the modal then the slide down animation does not work correctly. It would just disappear instead of sliding all the way down. This is the reason why the modal returns null for children if it's not visible. 
2. You're right, I'll get that removed.